modules/transactions/src/main/java/org/apache/ignite/internal/tx/views/TransactionsViewProvider.java

```diff
@@ -124,19 +110,15 @@ static class TxInfo {
         private final String type;
         private final String priority;
 
-        static TxInfo readOnly(UUID id) {
-            return new TxInfo(id, PENDING, true);
-        }
+        TxInfo(UUID id, TxStateMeta txStateMeta) {
```

Comment:
you don't need to pass `id` explicitly, this can be derived from InternalTransaction held by state